Add example for asserting with java on getting_started/using_selenium

harshitBhardwaj97 opened the pull request:
## **User description**
**Thanks for contributing to the Selenium site and documentation!**
**A PR well described will help maintainers to review and merge it quickly**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines.
Avoid large PRs, and help reviewers by making them as simple and short as possible.


<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Describe your changes in detail -->
Example for Asserting added for java language, by creating AssertionsTest.java file and 
adding relevant code lines reference in the markdown file.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.
<!--- Provide a general summary of your changes in the Title above -->


___

## **Type**
enhancement, documentation


___

## **Description**
- Added `AssertionsTest.java` demonstrating how to use assertions in Selenium tests with Java.
- Updated English and Japanese versions of the "Using Selenium" documentation to reference the new assertions example.
- Made minor formatting adjustments to the documentation for better readability and consistency.



___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>AssertionsTest.java</strong><dd><code>Add Java Selenium Assertions Example</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/java/src/test/java/dev/selenium/getting_started/AssertionsTest.java

<li>Added a new Java test class <code>AssertionsTest</code> with a test method <br><code>assertionTest</code> demonstrating how to perform assertions in Selenium <br>tests.<br> <li> Demonstrates assertions on URL, page title, and visibility of an <br>element.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1672/files#diff-e786f4209a9c867652708f57f29d6b29ce9fc2eb68799269c9224a1acd48bff5">+34/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>using_selenium.en.md</strong><dd><code>Update Selenium Getting Started Guide with Assertions Example</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/getting_started/using_selenium.en.md

<li>Updated the documentation to include a reference to the newly added <br><code>AssertionsTest.java</code>.<br> <li> Minor formatting adjustments for consistency.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1672/files#diff-f521d4cd05fea84ac0fd18184a214f8ac2cd023a7f2c1a7e2b43495504b7b9f0">+27/-14</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>using_selenium.ja.md</strong><dd><code>Update Japanese Selenium Documentation with Assertions Example</code></dd></summary>
<hr>
      
website_and_docs/content/documentation/webdriver/getting_started/using_selenium.ja.md

<li>Included reference to <code>AssertionsTest.java</code> in the Japanese version of <br>the documentation.<br> <li> Made similar formatting improvements as in the English version.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1672/files#diff-53fac7bfd25fb50de7731415fedc5ea636bcdaa83166f476b3c9e13a691a6c7c">+27/-14</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

harshitBhardwaj97 committed:
Add example for asserting with java on page getting_started/using_selenium/

Example for Asserting added for java language, by creating AssertionsTest.java
file and adding relevant code lines reference in the markdown file.

harshitBhardwaj97 committed:
Documentation updated for using_selenium.ja.md

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/seleniumhq.github.io?pullRequest=1672) <br/>All committers have signed the CLA.

harshitBhardwaj97 committed:
Add example of Setting up and Tear down with java

Example for Setting up and Tearing down added
for java language, by creating SetupAndTeardownTest.java
file, and adding relevant code lines reference in the markdown file(s).

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/SetupAndTeardownTest.java:
```diff
@@ -0,0 +1,78 @@
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class SetupAndTeardownTest {
+
+	private WebDriver driver;
+	private WebDriverWait wait;
+	private By projectsLink = By.xpath("//a/span[text()='Projects']");
+	private By searchFieldSpan = By.xpath("//span[text()='Search']");
+	private By searchField = By.id("docsearch-input");
+	private By searchResultsFirstItem = By.cssSelector("ul[role='listbox'] li");
+
+	@BeforeAll
+	public static void beforeAll() {
```
The script we want to show exists already: https://github.com/SeleniumHQ/seleniumhq.github.io/blob/trunk/examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java#L16.

Could you please link the existing one instead? Feel free to modify it to have the setup and tear down.

harshitBhardwaj97 replied:
![image](https://github.com/SeleniumHQ/seleniumhq.github.io/assets/75480921/a98d0e4a-76c0-4aaa-8e2f-cb05d8dfbc89)

Actually by reading this line, I implied that for Assertion and Setup / Teardown examples, we should be creating different files.
So based on the above comment, should I remove the SetupAndTeardownTest.java and AssertionsTest.java ? In that case, we can add the setup and teardown code in UsingSelenium.java and assertions are already being used in that.
 
Kindly let me know how you want to do it, I will make the appropriate changes.

diemol replied:
Just using the `UsingSelenium.java` is enough. Thank you.

harshitBhardwaj97 replied:
Got it, thanks for the clarification. Making changes and raising the PR.

harshitBhardwaj97 committed:
Assertion and Setup/Teardown examples linked

Assertion and Setup/Teardown examples linked with
existing UsingSeleniumTest.java, removed AssertionsTest.java
and SetupAndTeardownTest.java

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java:
```diff
@@ -1,38 +1,65 @@
+	@BeforeAll
+	public static void beforeAll() {
+		System.out.println("This method will be executed only once before all the tests.");
+	}
```
Why is this needed? I believe `BeforeEach` is enough.

harshitBhardwaj97 replied:
BeforeAll block removed.

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java:
```diff
@@ -1,38 +1,65 @@
+	@BeforeEach
+	public void setup() {
+		System.out.println(
+				"  This method will be executed before each test, so here write the common code that has to be executed.");
```
We prefer to avoid these comments, as this should be explained in the markdown file.

harshitBhardwaj97 replied:
Comment removed.

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java:
```diff
@@ -1,38 +1,65 @@
-        assertEquals("Received!", value);
+	@AfterEach
+	public void teardown() {
+		System.out.println("  This method will be executed after each test, so here write the clean up code.");
```
Same comment as above regarding comments.

harshitBhardwaj97 replied:
This comment also removed.

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java:
```diff
@@ -1,38 +1,65 @@
 
-        driver.quit();
-    }
+	@AfterAll
+	public static void afterAll() {
+		System.out.println("This method will be executed only once after all the tests.");
+	}
```
I belive we do not need this.

harshitBhardwaj97 replied:
AfterAll block removed.

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java:
```diff
@@ -1,38 +1,65 @@
+		System.out.println(
+				"  This method will be executed before each test, so here write the common code that has to be executed.");
+		driver = new ChromeDriver();
+		driver.manage().window().maximize();
```
We do not need this line, it was not in the original script.

harshitBhardwaj97 replied:
Sure, "driver.manage().window().maximize()" removed.

diemol commented on examples/java/src/test/java/dev/selenium/getting_started/UsingSeleniumTest.java:
```diff
@@ -1,38 +1,65 @@
+				"  This method will be executed before each test, so here write the common code that has to be executed.");
+		driver = new ChromeDriver();
+		driver.manage().window().maximize();
+		driver.manage().timeouts().implicitlyWait(Duration.ofMillis(500));
+		driver.get("https://www.selenium.dev/selenium/web/web-form.html");
```
This should be part of the test, not the setup.

harshitBhardwaj97 replied:
This code has been moved from setup to test body.

diemol reviewed (changes requested):
Could you please also make changes to the Portuguese and Chinese translations?

harshitBhardwaj97 committed:
Do the requested changes

1. BeforeAll and AfterAll functions removed.
2. Print statements removed.
3. driver.manage().window().maximize() removed.
4. Driver navigation and implicit wait code moved into test body.
5. Markdown changes done in all the 4 languages.

harshitBhardwaj97 commented:
As asked, the following changes have been done - 
1. BeforeAll and AfterAll functions removed.
2. Print statements removed.
3. driver.manage().window().maximize() removed.
4. Driver navigation and implicit wait code moved into test body.
5. Markdown changes done in all the 4 languages.

diemol reviewed (approved):
Thank you, @harshitBhardwaj97!

harshitBhardwaj97 commented:
> Thank you, @harshitBhardwaj97!
Welcome Sir, I will try to contribute more. Have a nice day.
